syft/frameworks/torch/tensors/interpreters/precision.py

```diff
@@ -215,90 +215,135 @@ def t(self, _self, *args, **kwargs):
 
         return response
 
-    def mul(self, other):
+    def mul_and_div(self, other, cmd):
         """
-        Hook manually mul to add the truncation part which is inherent to multiplication
-        in the fixed precision setting
+        Hook manually mul and div to add the trucation/rescaling part
+        which is inherent to these operations in the fixed precision setting
         """
-        if isinstance(other, (int, torch.Tensor)):
+        changed_sign = False
+        if isinstance(other, FixedPrecisionTensor):
+            assert (
+                self.precision_fractional == other.precision_fractional
+            ), "In mul and div, all args should have the same precision_fractional"
+            assert self.base == other.base, "In mul and div, all args should have the same base"
+
+        if isinstance(other, (int, torch.Tensor, AdditiveSharingTensor)):
             new_self = self.child
             new_other = other
 
         elif isinstance(self.child, (AdditiveSharingTensor, MultiPointerTensor)) and isinstance(
             other.child, torch.Tensor
         ):
-            # If we try to multiply a FPT>AST with a FPT>torch.tensor,
-            # we want to perform AST * torch.tensor
-            new_self = self.child
+            # If operands are FPT>AST and FPT>torch.tensor,
+            # we want to perform the operation on AST and torch.tensor
+            if cmd == "mul":
+                new_self = self.child
+            elif cmd == "div":
+                new_self = self.child * self.base ** self.precision_fractional
             new_other = other
 
         elif isinstance(other.child, (AdditiveSharingTensor, MultiPointerTensor)) and isinstance(
             self.child, torch.Tensor
         ):
-            # If we try to multiply a FPT>torch.tensor with a FPT>AST,
-            # we swap operators so that we do the same operation as above
-            new_self = other.child
-            new_other = self
-
-        elif isinstance(self.child, (AdditiveSharingTensor, MultiPointerTensor)) and isinstance(
-            other.child, (AdditiveSharingTensor, MultiPointerTensor)
-        ):
-            # If we try to multiply a FPT>torch.tensor with a FPT>AST,
+            # If operands are FPT>torch.tensor and FPT>AST,
             # we swap operators so that we do the same operation as above
-            new_self, new_other, _ = syft.frameworks.torch.hook_args.unwrap_args_from_method(
-                "mul", self, other, None
-            )
+            if cmd == "mul":
+                new_self = other.child
+                new_other = self
+            elif cmd == "div":
+                # TODO how to divide by AST?
+                raise NotImplementedError("Division by AST not implemented")
```

Comment:
"Division of a FPT by an AST not implemented" would be more clear
Also let's not use FPT & AST abbreviations in our debug messages, people might not know them